content/basics/basics.mdx

```diff
@@ -0,0 +1,327 @@
+makes `AnimalView` automatically re-render itself if anything that it observes
+changes. In this case the observable information is the computed property
+`isHungry`, which in turn it based on the observable property `energy`.
+`AnimalView` automatically re-renders itself only if `isHungry` changes. You
+don't need to do anything else to make that happen.
+
+When you wrap your React component in `observer`, it tracks any observable
+property or object you access during rendering it, and reacts to any changes to
+these. It won't react to any changes in observable properties you _don't_
+directly or indirectly use during rendering.
+
+## When do observers observe?
+
+Note that MobX can only observe properties, not simple values like a `string`
+or a `number`. So if you write your component to only receive simple values, it
+does not benefit from observability:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+// Uses observer() but does not benefit from observability
+const NonObservingAnimalView = observer(
+  ({
+    name,
+    species,
+    isHungry,
+  }: {
+    name: string
+    species: string
+    isHungry: boolean
+  }) => {
+    return (
+      <div>
+        {name} is of species {species} and is{' '}
+        {isHungry ? 'hungry' : 'not hungry'}.
+      </div>
+    )
+  },
+)
+```
+
+The component `NonObservingAnimalView` works perfectly if it is used within
+another component that receives an `Animal` instance and is also an observer,
+but it re-render each and every time, because MobX cannot see whether
+`name`, `species` or `isHungry` are changed. Our original `AnimalView` _does_
+observe it, because it can track property accesses.
+
+Wrapping a component in `observer` is cheap, so if you use MobX with react,
+it's recommended to just wrap all your components in `observer`, even if the
+component does not use any observable information yet. There is no need
+distinguish between React components that observe and do not observe in this.
+
+## Observable arrays
+
+MobX does support observable arrays, observable
+objects and observable maps, so those _can_ be observed for changes by
+a React component. We demonstrate that here with `animals`:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const AnimalsView = observer(({ animals }: { animals: Animal[] }) => {
+  return (
+    <ul>
+      {animals.map(animal => (
+        <ul>
+          <AnimalView key={animal.name} animal={animal} />
+        </ul>
+      ))}
+    </ul>
+  )
+})
+```
+
+Instances of our `Zoo` class have an `animals` property that is observable, so
+let's use that:
+
+```tsx
+const wilma = new Animal('Wilma', 'penguin')
+
+const zoo = new Zoo()
+zoo.addAnimal(fred)
+zoo.addAnimal(wilma)
+```
+
+We can now render our animals:
+
+```tsx
+<AnimalsView animals={zoo.animals} />
+```
+
+`AnimalsView` automatically re-renders itself whenever you change the array, in
```

Comment:
It would be worth to mention common pitfall here where people add `observer` around parent component rendering `AnimalsView` but to that component alone. In such a case, adding/removing from such array won't be reactive.